firefox, firefox-bin: 59.0.2 -> 59.0.3, firefox-esr: 2.7.3esr -> 52.7.4esr

taku0 opened the pull request:
###### Motivation for this change

- Just bumping version (fix for compatibility with Windows 10 April 2018 update)

https://www.mozilla.org/en-US/firefox/59.0.3/releasenotes/
https://www.mozilla.org/en-US/firefox/52.7.4/releasenotes/

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Tested only `firefox-bin`.

---

taku0 committed:
firefox-bin: 59.0.2 -> 59.0.3

taku0 committed:
firefox: 59.0.2 -> 59.0.3

taku0 committed:
firefox-esr: 52.7.3esr -> 52.7.4esr

andir commented:
Same issue for me. The downloads have always been slow but not as slow as this time.

The build fails with the same hash as for @vcunat  `2dh2hp307iz90969i8r96hdmd0d2v2d5zjp0n3klsn789sd6b3jdznb4mc5ghpm2rpcl48rqakw6apvmhidcahc1zbc1fq6j6a86jqp`.

According to http://releases.mozilla.org/pub/firefox/releases/59.0.3/SOURCE the commit hash we are using is correct for that specific tag (is that the right term with hg?). Also https://hg.mozilla.org/releases/mozilla-release/rev/FIREFOX_59_0_3_RELEASE points to the same.

taku0 commented:
Updated hash. I have no idea why the hash changed. The URL point to the `FIREFOX_59_0_3_RELEASE` tag and it should not change.

https://hg.mozilla.org/releases/mozilla-release/file/6b51784853e4

vcunat commented:
Looking at download links at https://hg.mozilla.org/releases/mozilla-release/rev/FIREFOX_59_0_3_RELEASE , I thought the tarball is a _dynamically_ generated one and thought we would better switch to `fetchzip` to avoid hash changes.  But I tried to re-fetch the 59.0 tarball, and the hash matched as we had it in nixpkgs, so maybe it was just some kind of "random" failure...

taku0 committed:
firefox: use fetchzip instead of fetchurl

taku0 commented:
Updated to use `fetchzip`.

andir commented:
Would you mind updating this for Firefox 60 which got released? I'll try to verify it again. Rebuilding the current state right now

taku0 commented:
OK, now updating. The archive server has source archive again, but `SHA512SUMS` have no hash for the source archive. I'm updating the nix expression manually.

taku0 committed:
firefox-bin: 59.0.3 -> 60.0

taku0 committed:
firefox: 59.0.3 -> 60.0

taku0 committed:
firefox-esr: 52.7.4esr -> 52.8.0esr, 60.0esr

taku0 commented:
Updated to 60.0. 

We have two ESR versions 52.8.0 and 60.0 for smooth migration.
https://www.mozilla.org/en-US/firefox/organizations/

andir commented:
Thanks!

What are your thoughts on the migration to version 60 as the default ESR version? IMHO on `master` we could do that right away but when backporting to relases (`18.03`…) we would keep the older ESR?

andir commented:
Also what is the current best practice in regards to target branch for this? Should this go into `staging` instead since we've a large number of rebuilds (caused by `nspr` & `nss`)? (cc @vcunat)

taku0 committed:
firefox-esr: make 60.x as the default

andir commented:
@taku0 looks good to me. Can you change the target branch to `staging`?

I've prepared a backport to 18.03 @ #40396.

andir committed:
nss: removed aarch64 patch

With the update to 3.36 the patch introduced with e96473b is no longer
required.

andir commented:
@taku0 you probably have to pick c716381380dd68be9989a513929e38653c5c559a since the aarch64 build is failing (on stable) with the newer nss.

Let us see what OfBorg says about it…

@GrahamcOfBorg build nss

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=a30f35a0-c846-42a7-a7dd-36c0d3434657)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
these paths will be fetched (3.08 MiB download, 11.47 MiB unpacked):
  /nix/store/9kmcfncsv6bl3cb161g2smwgwa1003pk-nss-3.36.1
  /nix/store/na74dbjsmani7v41r6jj0m04n5nfcfc3-nspr-4.19
  /nix/store/r6lpv1awkhjxpjbkl2gv2whzp9ydw302-gcc-7.3.0-lib
  /nix/store/y814z576i5za17a76r49m6vq7z0xkn8b-sqlite-3.22.0
copying path '/nix/store/r6lpv1awkhjxpjbkl2gv2whzp9ydw302-gcc-7.3.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/na74dbjsmani7v41r6jj0m04n5nfcfc3-nspr-4.19' from 'https://cache.nixos.org'...
copying path '/nix/store/y814z576i5za17a76r49m6vq7z0xkn8b-sqlite-3.22.0' from 'https://cache.nixos.org'...
copying path '/nix/store/9kmcfncsv6bl3cb161g2smwgwa1003pk-nss-3.36.1' from 'https://cache.nixos.org'...
/nix/store/9kmcfncsv6bl3cb161g2smwgwa1003pk-nss-3.36.1
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=88c80840-33fc-4108-9b17-fe7c4c560152)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/95v4pg2hfrbskb3jf3krpgikfjv6cgnr-apple-framework-NetFS.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/31zqh6hrchxqd32md2k6akxs41asn9wf-apple-framework-OpenDirectory.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/jnp3zyprgkmfb7iwdf6k2wa4syiqbg4g-apple-framework-CoreAudio.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/8h5wr1lcz6vrbm5jg81yd6mza9q8bj8q-apple-framework-DiskArbitration.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/nnbz3wivjk6z1dm2dx5y8z61ijpm7v9b-apple-framework-Security.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/35cn8lczfs8r7gsby7na1cilfa09yny2-apple-framework-ServiceManagement.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/phaw46bfmfxi5wxricsxcc8j5x92fm8c-apple-framework-CoreServices.drv': 12 dependencies couldn't be built
cannot build derivation '/nix/store/shvk6mmk1n24q0rp45kiw2mhyyb4rysk-nspr-4.19.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/j4a71jn6fmzn41h8scjh5n78b43p8ww3-nss-3.36.1.drv': 7 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/j4a71jn6fmzn41h8scjh5n78b43p8ww3-nss-3.36.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=6aae43b2-62b7-4fdd-8d22-159e5981f439)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/95v4pg2hfrbskb3jf3krpgikfjv6cgnr-apple-framework-NetFS.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/31zqh6hrchxqd32md2k6akxs41asn9wf-apple-framework-OpenDirectory.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/jnp3zyprgkmfb7iwdf6k2wa4syiqbg4g-apple-framework-CoreAudio.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/8h5wr1lcz6vrbm5jg81yd6mza9q8bj8q-apple-framework-DiskArbitration.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/nnbz3wivjk6z1dm2dx5y8z61ijpm7v9b-apple-framework-Security.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/35cn8lczfs8r7gsby7na1cilfa09yny2-apple-framework-ServiceManagement.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/phaw46bfmfxi5wxricsxcc8j5x92fm8c-apple-framework-CoreServices.drv': 12 dependencies couldn't be built
cannot build derivation '/nix/store/shvk6mmk1n24q0rp45kiw2mhyyb4rysk-nspr-4.19.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/j4a71jn6fmzn41h8scjh5n78b43p8ww3-nss-3.36.1.drv': 7 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/j4a71jn6fmzn41h8scjh5n78b43p8ww3-nss-3.36.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=4aff4ae6-fb76-4056-945c-f3eaf89cfe9d)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
these paths will be fetched (1.38 MiB download, 5.02 MiB unpacked):
  /nix/store/9kmcfncsv6bl3cb161g2smwgwa1003pk-nss-3.36.1
  /nix/store/na74dbjsmani7v41r6jj0m04n5nfcfc3-nspr-4.19
copying path '/nix/store/na74dbjsmani7v41r6jj0m04n5nfcfc3-nspr-4.19' from 'https://cache.nixos.org'...
copying path '/nix/store/9kmcfncsv6bl3cb161g2smwgwa1003pk-nss-3.36.1' from 'https://cache.nixos.org'...
/nix/store/9kmcfncsv6bl3cb161g2smwgwa1003pk-nss-3.36.1
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=beaea73b-11a3-4fda-b5e3-c36c99ce3b51)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
patching file nss/lib/ckfw/pem/psession.c
patching file nss/lib/ckfw/pem/pslot.c
patching file nss/lib/ckfw/pem/ptoken.c
patching file nss/lib/ckfw/pem/rsawrapr.c
patching file nss/lib/ckfw/pem/util.c
patching file lib/freebl/Makefile
Hunk #1 FAILED at 522.
1 out of 1 hunk FAILED -- saving rejects to file lib/freebl/Makefile.rej
builder for '/nix/store/c874ky3pm7ymyb4v7wc7n6708jkan2mv-nss-3.36.1.drv' failed with exit code 1
[31;1merror:[0m build of '/nix/store/c874ky3pm7ymyb4v7wc7n6708jkan2mv-nss-3.36.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=9af0bccc-0197-4363-8e77-1676fe4af298)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
patching file nss/lib/ckfw/pem/psession.c
patching file nss/lib/ckfw/pem/pslot.c
patching file nss/lib/ckfw/pem/ptoken.c
patching file nss/lib/ckfw/pem/rsawrapr.c
patching file nss/lib/ckfw/pem/util.c
patching file lib/freebl/Makefile
Hunk #1 FAILED at 522.
1 out of 1 hunk FAILED -- saving rejects to file lib/freebl/Makefile.rej
builder for '/nix/store/c874ky3pm7ymyb4v7wc7n6708jkan2mv-nss-3.36.1.drv' failed with exit code 1
[31;1merror:[0m build of '/nix/store/c874ky3pm7ymyb4v7wc7n6708jkan2mv-nss-3.36.1.drv' failed
```
</p></details>

taku0 commented:
Rebased onto `staging` branch. The branch already has `nspr-4.19` and `nss-3.36.1`.

vcunat commented:
aarch64-linux is probably OK now, but let me finish pre-testing the builds on aarch64.nixos.community.

andir commented:
@GrahamcOfBorg build nss

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=8740839d-f167-497d-be8e-b5d6fbab88e1)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/cqdr1xhvdq1jxjjfis37jvis4fqfl3s6-nss-3.36.1
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=296b14fc-2e8c-43a8-b098-71ac2f2d5b95)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/9kmcfncsv6bl3cb161g2smwgwa1003pk-nss-3.36.1
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40006&attempt_id=33ad4a55-4aa2-4e91-9c15-9b31700ad435)

Attempted: nss

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/jagbxb72m22vy9fhrhngycii2yjy56hi-apple-framework-NetFS.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/zaaj63j0lhfhfnzqy1w7bjs2z7l815x2-apple-framework-OpenDirectory.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/cy5xx2s0flif03rs5f25kxvp9ihmrh0i-apple-framework-CoreAudio.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/qpxsdk04iaq2dc6slfl1lank6x8z8mi2-apple-framework-DiskArbitration.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/zw7hp9xj62ghhxpx07m2941nvps11g0n-apple-framework-Security.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/yhi3zqr5iafb8kh673iir472lizd9i9v-apple-framework-ServiceManagement.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/qjmjz0l850zgsn6k1pkinvc5z1pnk76a-apple-framework-CoreServices.drv': 11 dependencies couldn't be built
cannot build derivation '/nix/store/p17l9qsvknfrwmbri387qp9qmscam78m-nspr-4.19.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/5nqzlyq49nl8l7ppcdsm6rdc4p2i4c9k-nss-3.36.1.drv': 5 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/5nqzlyq49nl8l7ppcdsm6rdc4p2i4c9k-nss-3.36.1.drv' failed
```
</p></details>

vcunat commented:
I've only now got to building firefox itself, after many packages like llvm and rustc.  We'll fix it up later in the worst case; there are far bigger breakages on staging ATM anyway.

danbst commented:
Guys, I'm disappointed about removal of `firefox-esr-unwrapped` top-level attribute. Not that it is bad, but that somehow jumped into `18.03` branch and broke stable API assumptions.

vcunat commented:
Right, I brought it back on 18.03.

ATM I'm not even certain it was intentional to remove it on master either, but intuitively if one wants ESR, one probably wants to switch the major version by hand (and thus avoid the default-esr names).
